src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Text.cs

```diff
@@ -155,30 +159,38 @@ internal static void ValidateFormatArgs(string name, CheckTypesContext checkType
             }
             else if (BinderUtils.TryGetConstantValue(checkTypesContext, args[1], out var formatArg))
             {
-                if (checkTypesContext.Features.PowerFxV1CompatibilityRules)
+                if (AllowedListToUseFormatString.Contains(argTypes[0]) || string.IsNullOrEmpty(formatArg))
```

Comment:
>if (AllowedListToUseFormatString.Contains(argTypes[0]) || string.IsNullOrEmpty(formatArg)) [](http://example.com/codeflow?start=16&length=90)

We should perform the check even if the format is not constant. For example, `Text(true, varFormat)` should be disallowed, or something like `Text("Hello", Match("d", "d").FullMatch)`.

Reply:
And we should only perform the check if PowerFxV1CompatibilityRules is true; legacy should not change.

Reply:
Updated to check this only when PowerFxV1CompatibilityRules is true. I add check at runtime that will cover when format is not constant.  and add tests for this case.